File: 56-MergeIntervals/56-MergeIntervals.kt
// Last updated: 5/4/2026, 8:09:04 PM
1class Solution {
2    fun removeElement(nums: IntArray, `val`: Int): Int {
3               for ((i, element) in nums.withIndex()) {
4        if (element == `val`) {
5            nums[i] = -1
6        }
7    }
8
9    var j: Int? = null
10    var i = 0
11
12    while (i < nums.size) {
13        if (nums[i] == -1 && j == null) {
14            j = i
15        }
16
17        if (nums[i] != -1 && j != null) {
18            nums[j] = nums[i]
19            nums[i] = -1
20            i = j
21            j = null
22        }
23
24        i++
25    }
26
27    return j ?: nums.size
28    }
29}

File: 56-MergeIntervals/56-MergeIntervals.rs
// Last updated: 5/4/2026, 8:39:39 PM
1impl Solution {
2    pub fn str_str(haystack: String, needle: String) -> i32 {
3        let h = haystack.as_bytes();
4        let n = needle.as_bytes();
5
6        if n.len() == 0 {
7            return 0;
8        }
9
10        if n.len() > h.len() {
11            return -1;
12        }
13
14        for i in 0..=h.len() - n.len() {
15            let mut match_found = true;
16
17            for j in 0..n.len() {
18                if h[i + j] != n[j] {
19                    match_found = false;
20                    break;
21                }
22            }
23
24            if match_found {
25                return i as i32;
26            }
27        }
28
29        -1
30    }
31}